dss/operations/__init__.py

```diff
@@ -0,0 +1,103 @@
+"""
+This file provides an interface to build the DSS operations CLI, parts of which may be
+executed "in the cloud", via Lambda-SQS integration.
+
+Commands are defined with a target-action model. While command arguments configured at the
+target level are moved into the action arguments.
+
+e.g
+```
+dispatch = DSSOperationsCommandDispatch()
+
+storage = dispatch.target("storage", arguments={"replica", dict(choices=["dev", "integration", "staging", "prod"])}
+
+@storage.action("verify-referential-integrity")
+def verify_ref_integrity(argv, args):
+    ...
+
+# execution:
+scripts/dss-ops.py storage verify-referential-integrity --replica staging
+```
+"""
+import os
+import argparse
+import logging
+import traceback
+from functools import partial
+from uuid import uuid4
+
+import dss
+from dss import Replica
+
+logger = logging.getLogger(__name__)
+if not os.environ.get("DSS_VERSION"):  # detect non-lambda environment
+    logger.setLevel(logging.DEBUG)
+    ch = logging.StreamHandler()
+    ch.setLevel(logging.DEBUG)
+    logger.addHandler(ch)
+
+class _target:
+    def __init__(self, target_name, dispatcher):
+        self.target_name = target_name
+        self.dispatch = dispatch
```

Comment:
Should this be `self.dispatch = dispatcher`?

Reply:
fixed

Reply:
fixed